mobx_codegen/test/util_test.dart

```diff
@@ -12,16 +12,16 @@ class MockMethod extends Mock implements MethodElement {}
 
 class MockType extends Mock implements DartType {}
 
+class DynamicMockType extends Mock implements DynamicType {}
+
 MockMethod mockFutureMethod({
   bool returnsDynamic = false,
   bool returnsFuture = false,
   bool returnsFutureOr = false,
   bool isAsync = false,
   bool isGenerator = false,
 }) {
-  final returnType = MockType();
-  // ignore: deprecated_member_use
-  when(() => returnType.isDynamic).thenReturn(returnsDynamic);
+  final returnType = returnsDynamic ? DynamicMockType() : MockType();
```

Comment:
This line is failing the static analysis when using the latest dependencies.
This static analysis bug was fixed in Dart 3.4.0 but mobx supports Dart 3.0.0 as a minimum. To keep supporting this version it can be simply changed to an if else block. Source: https://github.com/dart-lang/sdk/issues/60255

```dart
  final DartType returnType;
  if (returnsDynamic) {
    returnType = DynamicMockType();
  } else {
    returnType = MockType();
  }
```